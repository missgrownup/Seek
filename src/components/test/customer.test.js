import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import Customer from '../customer'
import mockCustomers from './data/mockCustomers'

configure({ adapter: new Adapter() })

const mockSelectedCustomer = {
	id: 'customer3',
	name: 'Alix Coffee Roasters',
	pricingRules: [
		{
			type: 'discount',
			prerequisite: { product2: 1 },
			result: 299.99,
		},
	],
}

describe('<Customer />', () => {
	it('should render customer component', () => {
		const wrapper = renderer
			.create(
				<Customer
					customers={mockCustomers}
					selectedCustomer={mockSelectedCustomer}
					handleCustomerChange={() => {}}
				/>,
			)
			.toJSON()
		expect(wrapper).toMatchSnapshot()
	})
})
