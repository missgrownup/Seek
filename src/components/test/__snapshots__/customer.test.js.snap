// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Customer /> should render customer component 1`] = `
<select
  onChange={[Function]}
  value="customer3"
>
  <option
    disabled={true}
    value=""
  >
    Select customer
  </option>
  <option
    value="customer1"
  >
    Default
  </option>
  <option
    data-pricingrules="[{\\"type\\":\\"mixnmatch\\",\\"prerequisite\\":{\\"product1\\":3},\\"result\\":{\\"product1\\":2}}]"
    value="customer2"
  >
    SecondBite
  </option>
  <option
    data-pricingrules="[{\\"type\\":\\"discount\\",\\"prerequisite\\":{\\"product2\\":1},\\"result\\":299.99}]"
    value="customer3"
  >
    Alix Coffee Roasters
  </option>
  <option
    data-pricingrules="[{\\"type\\":\\"mixnmatch\\",\\"prerequisite\\":{\\"product2\\":5},\\"result\\":{\\"product2\\":4}},{\\"type\\":\\"discount\\",\\"prerequisite\\":{\\"product3\\":1},\\"result\\":389.99}]"
    value="customer4"
  >
    Myer
  </option>
</select>
`;

exports[`Customer component should render customer component 1`] = `
<select
  onChange={[Function]}
  value="customer3"
>
  <option
    disabled={true}
    value=""
  >
    Select customer
  </option>
  <option
    value="customer1"
  >
    Default
  </option>
  <option
    data-pricingrules="[{\\"type\\":\\"mixnmatch\\",\\"prerequisite\\":{\\"product1\\":3},\\"result\\":{\\"product1\\":2}}]"
    value="customer2"
  >
    SecondBite
  </option>
  <option
    data-pricingrules="[{\\"type\\":\\"discount\\",\\"prerequisite\\":{\\"product2\\":1},\\"result\\":299.99}]"
    value="customer3"
  >
    Alix Coffee Roasters
  </option>
  <option
    data-pricingrules="[{\\"type\\":\\"mixnmatch\\",\\"prerequisite\\":{\\"product2\\":5},\\"result\\":{\\"product2\\":4}},{\\"type\\":\\"discount\\",\\"prerequisite\\":{\\"product3\\":1},\\"result\\":389.99}]"
    value="customer4"
  >
    Myer
  </option>
</select>
`;
